### PR TITLE
[fa] Add missing translations

### PR DESCRIPTION
**Pull request by msalehi-d**

### CODE REVIEW

**Comment by kenjis** [comment]
Thank you!

But it seems there are some missing translations.
Can you fix?

```
There were 2 failures:

1) Translations\Tests\FarsiTranslationTest::testAllConfiguredLanguageFilesAreTranslated with data set "fa" ('fa')
Failed asserting that language files "CLI.php", "Cast.php", "Cookie.php", "Encryption.php", "Fabricator.php", "Filters.php", "Publisher.php", "RESTful.php", "Security.php", "Test.php" in the main repository are translated in "fa" locale.
Failed asserting that an array is empty.

/home/runner/work/translations/translations/tests/Language/AbstractTranslationTestCase.php:108
phpvfscomposer:///home/runner/work/translations/translations/vendor/phpunit/phpunit/phpunit:97

2) Translations\Tests\FarsiTranslationTest::testAllConfiguredLanguageKeysAreIncluded with data set "fa" ('fa')
Failed asserting that the language keys "Core.enabledZlibOutputCompression", "Core.invalidPhpVersion", "Database.emptyPrimaryKey", "Database.fieldNotExists", "Database.forEmptyInputGiven", "Database.forFindColumnHaveMultipleColumns", "Database.methodNotAvailable", "Database.noDateFormat", "Database.noPrimaryKey", "Database.parseStringFail", "Database.tableNotFound", "Email.exitStatus", "Files.expectedDirectory", "Files.expectedFile", "Format.invalidFormatter", "Format.invalidMime", "HTTP.controllerNotFound", "HTTP.emptyController", "HTTP.invalidSameSiteSetting", "HTTP.methodNotFound", "Images.fileNotSupported", "Images.webpNotSupported", "Log.invalidMessageType", "Migrations.added", "Migrations.batch", "Migrations.batchNotFound", "Migrations.filename", "Migrations.generalFault", "Migrations.group", "Migrations.latest", "Migrations.migCreate", "Migrations.migHelpCurrent", "Migrations.migHelpLatest", "Migrations.migHelpRefresh", "Migrations.migHelpRollback", "Migrations.migHelpSeed", "Migrations.migHelpVersion", "Migrations.migInvalidVersion", "Migrations.migMissingSeeder", "Migrations.migNumberError", "Migrations.migSeeder", "Migrations.migrated", "Migrations.nameMigration", "Migrations.nameSeeder", "Migrations.namespace", "Migrations.noneFound", "Migrations.on", "Migrations.refreshConfirm", "Migrations.removed", "Migrations.rollBackConfirm", "Migrations.rollingBack", "Migrations.toVersion", "Migrations.toVersionPH", "Migrations.version", "Router.invalidControllerName", "Router.invalidDynamicController", "Session.invalidSameSiteSetting", "Time.invalidFormat", "Validation.alpha_numeric_punct", "Validation.equals", "Validation.hex", "Validation.is_not_unique", "Validation.not_equals", "Validation.not_in_list", "Validation.string", "View.invalidCellClass", "View.invalidDecoratorClass", "View.noCellClass", "View.tagSyntaxError" in the main repository are included for translation in "fa" locale.
Failed asserting that an array is empty.

/home/runner/work/translations/translations/tests/Language/AbstractTranslationTestCase.php:166
phpvfscomposer:///home/runner/work/translations/translations/vendor/phpunit/phpunit/phpunit:97
```

**Comment by msalehi-d** [comment]
It may take a while, but I'll work on it.

**Review by totoprayogo1916 — COMMENTED** [review]
`Language/Cast.php` should be removed

**Comment by msalehi-d** [comment]
> `Language/Cast.php` should be removed

Thanks. It still exists in the CI4 fresh installation.

**Comment by paulbalandan** [comment]
> > `Language/Cast.php` should be removed
> 
> Thanks. It still exists in the CI4 fresh installation.

@totoprayogo1916 meant was to move `Language/Cast.php` to `Language/fa/Cast.php`

File: Language/fa/Cookie.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+// Cookie language settings
+return [
+    'invalidExpiresTime'    => 'نوع داده {0} برای ویژگی "Expires" معتبر نیست. نوع داده های معتبر: string, integer, DateTimeInterface object.
+',
```
Comment (paulbalandan) [inline Language/fa/Cookie.php:15]:
```suggestion
    'invalidExpiresTime'    => 'نوع داده {0} برای ویژگی "Expires" معتبر نیست. نوع داده های معتبر: string, integer, DateTimeInterface object.',
```

File: Language/fa/Validation.php [hunk 2]
```diff
@@ -54,8 +61,8 @@
     'valid_date'            => 'فیلد {field} باید یک تاریخ معتبر باشد.',
 
     // Credit Cards
-    'valid_cc_num' => '{field} یک شماره‌ی کارت اعتباری معتبر نیست.',
-
+    'valid_cc_num' => 'فیلد {field} به نظر یک شماره کارت اعتباری معتبر نیست.',
+
```
Comment (paulbalandan) [inline Language/fa/Validation.php:65]: Remove trailing whitespace.
```suggestion
```

File: Language/fa/View.php [hunk 3]
```diff
@@ -14,4 +14,7 @@
     'invalidCellMethod'     => '{class}::{method} تابع معتبری نیست.',
     'missingCellParameters' => '{class}::{method} پارامتری ندارد.',
     'invalidCellParameter'  => '{0} نام پارامتر معتبری نیست.',
-];
+    'noCellClass'           => 'هیچ کلاس view cell مشخص نشده.',
+    'invalidCellClass'      => 'کلاس view cell با نام {0} پیدا نشد.',
+    'tagSyntaxError'        => 'یک خطای گرامی در Parser tags وجود دارد: {0}',
+    'invalidDecoratorClass' => '{0} یک View Decorator معتبر نیست.',];
```
Comment (paulbalandan) [inline Language/fa/View.php:20]:
```suggestion
    'invalidDecoratorClass' => '{0} یک View Decorator معتبر نیست.',
];
```

**Review by paulbalandan — CHANGES_REQUESTED** [review]

**Comment by msalehi-d** [comment]
Why all changes for most of the files are gone? Last update is 12 months ago

**Comment by paulbalandan** [comment]
It seems the maintainer for this locale is not active. If you have time, you can replace as maintainer.

**Comment by msalehi-d** [comment]
@paulbalandan Yes please. I'll upload files again after that

File: Language/fa/Migrations.php [hunk 4]
```diff
@@ -13,11 +13,46 @@
 return [
     // Migration Runner
     'missingTable' => 'جدول مایگریشن تنظیم نشده است.',
```
Comment (paulbalandan) [inline Language/fa/Migrations.php:15]:
```suggestion
    'missingTable'  => 'جدول مایگریشن تنظیم نشده است.',
```

File: Language/fa/Router.php [hunk 5]
```diff
@@ -13,4 +13,7 @@
 return [
     'invalidParameter'    => 'یک پارامتر با مقدار مورد انتظار تطابق ندارد.',
     'missingDefaultRoute' => 'امکان تشخیص محتوایی برای نمایش وجود ندارد. مسیر پیش فرض در فایل روتینگ مشخص نشده است.',
+    'invalidDynamicController' => 'کنترلر داینامیک به دلایل امنیتی مجاز نیست.  Route handler: {0}',
+    'invalidControllerName'    => 'برای جدا کننده‌ی فضای نام از بک اسلش (\) استفاده کنید، نه اسلش (/). Route handler: {0}',
+
```
Comment (paulbalandan) [inline Language/fa/Router.php:18]:
```suggestion
    'invalidParameter'         => 'یک پارامتر با مقدار مورد انتظار تطابق ندارد.',
    'missingDefaultRoute'      => 'امکان تشخیص محتوایی برای نمایش وجود ندارد. مسیر پیش فرض در فایل روتینگ مشخص نشده است.',
    'invalidDynamicController' => 'کنترلر داینامیک به دلایل امنیتی مجاز نیست.  Route handler: {0}',
    'invalidControllerName'    => 'برای جدا کننده‌ی فضای نام از بک اسلش (\) استفاده کنید، نه اسلش (/). Route handler: {0}',
```

File: Language/fa/HTTP.php [hunk 6]
```diff
@@ -1,3 +1,4 @@
+
 <?php
```
Comment (paulbalandan) [inline Language/fa/HTTP.php:2]:
```suggestion
<?php
```

File: Language/fa/Validation.php [hunk 7]
```diff
@@ -54,8 +61,7 @@
     'valid_date'            => 'فیلد {field} باید یک تاریخ معتبر باشد.',
 
     // Credit Cards
-    'valid_cc_num' => '{field} یک شماره‌ی کارت اعتباری معتبر نیست.',
-
+    'valid_cc_num' => 'فیلد {field} به نظر یک شماره کارت اعتباری معتبر نیست.',
```
Comment (paulbalandan) [inline Language/fa/Validation.php:64]:
```suggestion
    'valid_cc_num' => 'فیلد {field} به نظر یک شماره کارت اعتباری معتبر نیست.',

```

File: Language/fa/Cast.php [hunk 8]
```diff
@@ -0,0 +1,24 @@
+
+<?php
```
Comment (paulbalandan) [inline Language/fa/Cast.php:2]:
```suggestion
<?php
```

**Review by paulbalandan — APPROVED** [review]

**Comment by paulbalandan** [comment]
Thank you, @msalehi-d

**Pull request merged**